api/src/main/java/me/lokka30/treasury/api/permission/node/Node.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/ArcanePlugins/Treasury>.
+ */
+
+package me.lokka30.treasury.api.permission.node;
+
+import me.lokka30.treasury.api.permission.context.Context;
+import me.lokka30.treasury.api.permission.node.type.NodeType;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public interface Node<Data> {
+
+    @NotNull NodeType<Data> nodeType();
+
+    @NotNull Context context();
+
+    @Nullable Integer weight();
```

Comment:
I would rather use the primitive instead of the wrapper type of int and set the default value to `-1`, or we could make a WeightBuilder/NodeWeightBuilder to separate the actual weight and the permission node context.

Reply:
I first had it with the primitive, but then told myself how are we gonna define no weight presence? Using -1 is not really the way, because some permission plugins may opt in to use such weight.

Reply:
true, just thought if it could work

Reply:
I've another idea. Why not use `int` anyway? We can allow permission plugins to return the default weight for a permission group if no weight is explicitly defined.

This way, client plugins can use and rely on the permissions API by working with effective weights, rather than merely the weights configured. If weights aren't configured for a specific permissions group, the default weight may be returned.

Reply:
I'm really 50/50 on this. More opinions wanted.

Reply:
I agree with A2 here.

Reply:
@Jikoo @MrNemo64 thoughts?

Reply:
@Jikoo @MrNemo64 thoughts?

Reply:
I still believe a primitive is definitely way the go. Using the wrapper type here would unnecessarily complicate the permission weights system.

Reply:
I think that a primitive is also the way to go - no weight should likely be handled the same way a weight of 0 would be for any calculations, so we're just forcing anyone doing those calculations to unwrap it.